jf_agent/bb_download.py

```diff
@@ -19,34 +37,82 @@ def _normalize_user(user):
 
 
 def get_all_users(client):
-    print('downloading bitbucket users... ✓')
+    print('downloading bitbucket users... ', end='', flush=True)
+    users = [_normalize_user(user) for user in client.admin.users]
+    print('✓')
+    return users
 
-    return [_normalize_user(user) for user in client.admin.users]
 
+def _normalize_project(api_project, redact_names_and_urls):
+    return {
+        'id': api_project['id'],
+        'login': api_project['key'],
+        'name': (
+            api_project.get('name')
+            if not redact_names_and_urls
+            else _project_redactor.redact_name(api_project.get('name'))
+        ),
+        'url': api_project['links']['self'][0]['href'] if not redact_names_and_urls else None,
```

Comment:
Why does the absence of `readact_name_and_urls` return None in the url case, and the un-redacted(?) string in the name case?

Reply:
Because names are just labels, and we can show "Redacted-0004" in the app, where the URL is used for generating a link, and so the app needs to know that there is no link (vs generating a link to a URL that doesn't actually exist).